<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Generative Models for Discrete Data 广义模型和离散数据 | 读Modern Statistics for Modern Biology随记</title>
  <meta name="description" content="学习Modern Statistics for Modern Biology记录" />
  <meta name="generator" content="bookdown 0.12 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Generative Models for Discrete Data 广义模型和离散数据 | 读Modern Statistics for Modern Biology随记" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="学习Modern Statistics for Modern Biology记录" />
  <meta name="github-repo" content="ecowang/M for M" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Generative Models for Discrete Data 广义模型和离散数据 | 读Modern Statistics for Modern Biology随记" />
  
  <meta name="twitter:description" content="学习Modern Statistics for Modern Biology记录" />
  

<meta name="author" content="王强" />


<meta name="date" content="2019-09-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html">
<link rel="next" href="statistical-modeling.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">读Modern Statistics for Modern Biology随记</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction 引言</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#section-1.1"><i class="fa fa-check"></i><b>1.1</b> 缘起</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#section-1.2"><i class="fa fa-check"></i><b>1.2</b> 原书目标</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#section-1.3"><i class="fa fa-check"></i><b>1.3</b> 本书特点</a><ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#section-1.3.1"><i class="fa fa-check"></i><b>1.3.1</b> 统计江湖的套路</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#section-1.3.2"><i class="fa fa-check"></i><b>1.3.2</b> 统计的常见概念和生命学科特殊性</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#section-1.3.3"><i class="fa fa-check"></i><b>1.3.3</b> 原书中主要内容</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#section-1.4"><i class="fa fa-check"></i><b>1.4</b> 需要的计算机技能</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Generative Models for Discrete Data 广义模型和离散数据</a><ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#section-2.1"><i class="fa fa-check"></i><b>2.1</b> 本章概述</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#section-2.2"><i class="fa fa-check"></i><b>2.2</b> 回顾一些基础知识</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#section-2.2.1"><i class="fa fa-check"></i><b>2.2.1</b> 概率论与数理统计</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#section-2.2.2"><i class="fa fa-check"></i><b>2.2.2</b> 生命科学一些基础知识（直接看图吧）</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#section-2.3"><i class="fa fa-check"></i><b>2.3</b> 本章主要内容</a><ul>
<li class="chapter" data-level="2.3.1" data-path="intro.html"><a href="intro.html#section-2.3.1"><i class="fa fa-check"></i><b>2.3.1</b> 大体框架</a></li>
<li class="chapter" data-level="2.3.2" data-path="intro.html"><a href="intro.html#section-2.3.2"><i class="fa fa-check"></i><b>2.3.2</b> 分头看看几种分布</a></li>
<li class="chapter" data-level="2.3.3" data-path="intro.html"><a href="intro.html#a-generative-model-for-epitope-detection"><i class="fa fa-check"></i><b>2.3.3</b> 应用一个表位检测模型（a generative model for epitope detection）</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistical-modeling.html"><a href="statistical-modeling.html"><i class="fa fa-check"></i><b>3</b> Statistical Modeling</a></li>
<li class="chapter" data-level="4" data-path="high-quality-graphics-in-r.html"><a href="high-quality-graphics-in-r.html"><i class="fa fa-check"></i><b>4</b> High-Quality Graphics in R</a></li>
<li class="chapter" data-level="5" data-path="mixture-models.html"><a href="mixture-models.html"><i class="fa fa-check"></i><b>5</b> Mixture Models</a></li>
<li class="chapter" data-level="6" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>6</b> Clustering</a></li>
<li class="chapter" data-level="7" data-path="testing.html"><a href="testing.html"><i class="fa fa-check"></i><b>7</b> Testing</a></li>
<li class="chapter" data-level="8" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html"><i class="fa fa-check"></i><b>8</b> Multivariate Analysis</a></li>
<li class="chapter" data-level="9" data-path="high-throughput-count-data.html"><a href="high-throughput-count-data.html"><i class="fa fa-check"></i><b>9</b> High-Throughput Count Data</a></li>
<li class="chapter" data-level="10" data-path="multivariate-methods-for-heterogeneous-data.html"><a href="multivariate-methods-for-heterogeneous-data.html"><i class="fa fa-check"></i><b>10</b> Multivariate Methods for Heterogeneous Data</a></li>
<li class="chapter" data-level="11" data-path="networks-and-trees.html"><a href="networks-and-trees.html"><i class="fa fa-check"></i><b>11</b> Networks and Trees</a></li>
<li class="chapter" data-level="12" data-path="image-data.html"><a href="image-data.html"><i class="fa fa-check"></i><b>12</b> Image Data</a></li>
<li class="chapter" data-level="13" data-path="supervised-learning.html"><a href="supervised-learning.html"><i class="fa fa-check"></i><b>13</b> Supervised Learning</a></li>
<li class="chapter" data-level="14" data-path="design-of-high-throughput-experiments-and-their-analyses.html"><a href="design-of-high-throughput-experiments-and-their-analyses.html"><i class="fa fa-check"></i><b>14</b> Design of High-Throughput Experiments and Their Analyses</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/ecowang/MforM" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">读Modern Statistics for Modern Biology随记</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Generative Models for Discrete Data 广义模型和离散数据</h1>
<div id="section-2.1" class="section level2">
<h2><span class="header-section-number">2.1</span> 本章概述</h2>
<p>这一章是假设各种符合某分布的数据，主要以DNA、碱基等为描述对象。原书中的说法是 top-bottom 的方式，从我们了解的各种概率分布（top）开始，设计符合这些分布原则的具体问题，如碱基分布模式等。当然这是一种理想化的训练，实际的问题多是反方向 bottom-top 的，通过分析实际数据获取数据规律，或者建立模型。</p>
<p>本章主要目标包括：</p>
<ul>
<li><p>获取特定模型下所有可能结果的概率，并将理论值与实际数据比较。
Learn how to obtain the probabilities of all possible outcomes from a given model and see how we can compare the theoretical frequencies with those observed in real data.</p></li>
<li><p>通过一个完整示例，了解如何应用泊松分布分析抗原数据。<br />
Explore a complete example of how to use the Poisson distribution to analyse data on epitope detection.</p></li>
<li><p>如何应用泊松分布、二项式分布、多项式分布模型来做实验。<br />
See how we can experiment with the most useful generative models for discrete data: Poisson, binomial, multinomial.</p></li>
<li><p>使用R函数计算概率和偶发事件。<br />
Use the R functions for computing probabilities and counting rare events.</p></li>
<li><p>生成符合特定分布的随机数。<br />
Generate random numbers from specified distributions.</p></li>
</ul>
</div>
<div id="section-2.2" class="section level2">
<h2><span class="header-section-number">2.2</span> 回顾一些基础知识</h2>
<div id="section-2.2.1" class="section level3">
<h3><span class="header-section-number">2.2.1</span> 概率论与数理统计</h3>
<p>看这一章，感觉有些概念不清楚，再去学陈希孺老先生的的《概率论与统计统计》。</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-8"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/ZJuhzAKtSo+j8wR5zY04RA.jpg" alt="陈希孺老先生的概率论与数理统计" width="60%" />
<p class="caption">
Figure 2.1: 陈希孺老先生的概率论与数理统计
</p>
</div>
<p>带着点问题重新硬着头皮读，才发现这真的如同很多牛人推荐的，是一本神仙书。原来读的时候太功利，总想翻上几页就能拿去在R语言上套函数解决实际问题了，可看了看都是理论性的东西用不上，就丢到一边。现在再看，才发现老先生真是不紧不慢、娓娓道来，从“道”的高度把概率论和数理统计的有关概念、发展历程、应用场景等等系统的解释一遍。虽然是入门书，但好多领域的拓展也有谈及，用平实的语言勾画了宏大的场景，让人越读越有跪感。</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>名词</th>
<th>定义</th>
<th>原书涉及</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>试验/Trial</td>
<td>要有明确界限，宽泛的可以包含“观察”。全部结果是试验前即明确的，如抛硬币2面，掷骰子6点，事件发生概率为0到正无穷。</td>
<td>Tossing a coin has two possible outcomes. This simple experiment, called a Bernoulli <strong>trial</strong>, is modeled using a so-called Bernoulli random variable.</td>
</tr>
<tr class="even">
<td>事件/envent</td>
<td>一个明确的陈述，界定了试验全部可能结果中一确定的部分，这个陈述或这一确定的部分，即一个事件。</td>
<td>In molecular biology, many situations involve counting <strong>events</strong>.</td>
</tr>
</tbody>
</table>
<p>古典概率计算归结为计算两个数M和N，这种计算多涉及排列组合。如n个不同物体，取r个的排列数是 <span class="math inline">\(P_r^n = n(n-1)(n-2)...(n-r-1)\)</span>，而不同组合数则为<span class="math inline">\(C_r^m = P_r^n/r!=n!/(r!()n-r)!\)</span> ，这个 <span class="math inline">\(C_r^m\)</span> 又称为<strong>组合系数</strong>，统计上习惯用这种形式来表示：<span class="math inline">\(C_r^n = \binom{n}{r}\)</span>。</p>
<p>组合系数 <span class="math inline">\(C_r^m\)</span> 又常称为<strong>二项式系数</strong>，因为常出现在下列二项式展开的公式中：
<span class="math display">\[(a + b)^n = \sum_{i=0}^{n} \binom{n}{i} a^ib^{n-i} \]</span>
还有<strong>多项式系数</strong>，n个相异物分成k堆，各堆个数分别为<span class="math inline">\(r_1 , r_2..., r_k\)</span>分法是。
<span class="math display">\[n!/(r_1!...r_k!)\]</span>
它是<span class="math inline">\((x_1 + x_2 + ... + x_k)^n\)</span>的展开式中，<span class="math inline">\(x_1^{r_1}...x_k^{r_k}\)</span>这一项的系数。</p>
<p>虽然不知道这几个公式和二项分布、多项分布有啥关系，不过老先生不会随便说，先放着慢慢领会吧。</p>
</div>
<div id="section-2.2.2" class="section level3">
<h3><span class="header-section-number">2.2.2</span> 生命科学一些基础知识（直接看图吧）</h3>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-9"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/DNA%20sequence.png" alt="DNA 展开结构示意图。染色体-染色质-组织蛋白-脱氧核糖核酸双螺旋结构-基因-碱基对" width="80%" />
<p class="caption">
Figure 2.2: DNA 展开结构示意图。染色体-染色质-组织蛋白-脱氧核糖核酸双螺旋结构-基因-碱基对
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-10"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/codon.jpg" alt="RNA上的密码子[codons](https://en.wikipedia.org/wiki/DNA_codon_table)，“how many codons use a certain spelling”" width="100%" />
<p class="caption">
Figure 2.3: RNA上的密码子<a href="https://en.wikipedia.org/wiki/DNA_codon_table">codons</a>，“how many codons use a certain spelling”
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-11"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/dna.jpg" alt="[In DNA sequencing, a read is an inferred sequence of base pairs (or base pair probabilities) corresponding to all or part of a single DNA fragment. A typical sequencing experiment involves fragmentation of the genome into millions of molecules, which are size-selected and ligated to adapters. The set of fragments is referred to as a sequencing library, which is sequenced to produce a set of reads.](https://en.wikipedia.org/wiki/Read_(biology))" width="100%" />
<p class="caption">
Figure 2.4: <a href="https://en.wikipedia.org/wiki/Read_(biology)">In DNA sequencing, a read is an inferred sequence of base pairs (or base pair probabilities) corresponding to all or part of a single DNA fragment. A typical sequencing experiment involves fragmentation of the genome into millions of molecules, which are size-selected and ligated to adapters. The set of fragments is referred to as a sequencing library, which is sequenced to produce a set of reads.</a>
</p>
</div>
</div>
</div>
<div id="section-2.3" class="section level2">
<h2><span class="header-section-number">2.3</span> 本章主要内容</h2>
<div id="section-2.3.1" class="section level3">
<h3><span class="header-section-number">2.3.1</span> 大体框架</h3>
<p>如原文所说，这一章是一种“从上而下”的套路练习。通过设计一些符合概率分布模型的数据，了解数据和概率之间的关系。主要讲了4种分布：</p>
<ul>
<li>伯努利分布（<strong>Bernoulli distribution</strong>），又称两点分布或者’0-1’分布，要么成功是1，要么失败是0，只有一种可能。</li>
</ul>
<blockquote>
<p>It is a special case of the binomial distribution for n = 1. In other words, it is a binomial distribution with a single trial (e.g. a single coin toss)。<br />
The Bernoulli distribution is closely related to the Binomial distribution. As long as each individual Bernoulli trial is independent, then the number of successes in a series of Bernoulli trails has a Binomial Distribution.The Bernoulli distribution can also be defined as the Binomial distribution with n = 1.<br />
Bernoulli distributions are used in logistic regression to model disease occurrence.<br />
<a href="https://www.statisticshowto.datasciencecentral.com/bernoulli-distribution/">Bernoulli distributions are used in logistic regression to model disease occurrence.</a>.</p>
</blockquote>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-12"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/bonuli.png" alt="伯努利分布的维基百科解释" width="100%" />
<p class="caption">
Figure 2.5: 伯努利分布的维基百科解释
</p>
</div>
<ul>
<li>二项式分布（<strong>binomial distribution</strong>），二项分布即重复n次的伯努里试验。在每次试验中只有两种可能的结果，而且是互相对立的，是独立的,与其它各次试验结果无关，结果事件发生的概率在整个系列试验中保持不变，则这一系列试验称为伯努力试验。<br />
二项分布的典型例子是扔硬币，硬币正面朝上概率为p, 重复扔n次硬币，k次为正面的概率即为一个二项分布概率。
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-13"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/binomial.png" alt="二项分布的维基百科解释" width="100%" />
<p class="caption">
Figure 2.6: 二项分布的维基百科解释
</p>
</div></li>
<li>泊松分布（<strong>Poisson distribution</strong>），泊松分布是二项分布n很大而p很小时的一种极限形式二项分布是说，已知某件事情发生的概率是p，那么做n次试验，事情发生的次数就服从于二项分布。泊松分布是指某段连续的时间内某件事情发生的次数，而且“某件事情”发生所用的时间是可以忽略的。例如，在五分钟内，电子元件遭受脉冲的次数，就服从于泊松分布。假如你把“连续的时间”分割成无数小份，那么每个小份之间都是相互独立的。在每个很小的时间区间内，电子元件都有可能“遭受到脉冲”或者“没有遭受到脉冲”，这就可以被认为是一个p很小的二项分布。而因为“连续的时间”被分割成无穷多份，因此n(试验次数)很大。所以，泊松分布可以认为是二项分布的一种极限形式。因为二项分布其实就是一个最最简单的“发生”与“不发生”的分布，它可以描述非常多的随机的自然界现象，因此其极限形式泊松分布自然也是非常有用的。
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-14"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/bosong.png" alt="泊松分布的维基百科解释" width="100%" />
<p class="caption">
Figure 2.7: 泊松分布的维基百科解释
</p>
</div></li>
<li>多项分布（<strong>multinomial distribution</strong>）二项扩展为多项就得到了多项分布。二项分布的试验结果只有两个(成功和失败)，而多项分布的试验结果则多于两个。如果试验的结果有三个，则是三项分布；如果结果有六个，则是六项分布比如扔骰子，不同于扔硬币，骰子有6个面对应6个不同的点数，这样单次每个点数朝上的概率都是1/6（对应p1~p6，它们的值不一定都是1/6，只要和为1且互斥即可
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-15"></span>
<img src="static/Users/eco/M%20for%20M/01-generative_files/multi.png" alt="多项式分布的wikipedia解释" width="100%" />
<p class="caption">
Figure 2.8: 多项式分布的wikipedia解释
</p>
</div></li>
</ul>
</div>
<div id="section-2.3.2" class="section level3">
<h3><span class="header-section-number">2.3.2</span> 分头看看几种分布</h3>
<div id="section-2.3.2.1" class="section level4">
<h4><span class="header-section-number">2.3.2.1</span> 泊松分布</h4>
<p>原书首先介绍了泊松分布的一个示例。在一段HIV基因上，一次转录发生突变的概率为<span class="math inline">\(5 × 10^{-4}\)</span>, 也就是说，一段含有10000个核苷酸（nucleotides）的HIV基因，在一次复制过程中，发生突变的核苷酸个数在5左右，标准差为<span class="math inline">\(\sqrt5\)</span>。这个5，在R语言的泊松分布公式中，称为频率参数（rate parameter），写作<span class="math inline">\(\lambda\)</span>。 通过确定了出现频率的泊松分布，可以计算某个频率出现的概率。如，计算这段DNA中出现3个变异的概率是多少，实现方式如下。</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">dpois</span>(<span class="dt">x =</span> <span class="dv">3</span>, <span class="dt">lambda =</span> <span class="dv">5</span>)</a></code></pre></div>
<pre><code>## [1] 0.1403739</code></pre>
<p>不太明白，lambda是平均数，这里是5，可能没有明确在10000里面发生5啊，那么<code>dpois(x = 30, lambda = 50)</code> 应该得到一样的值吧，可实际上差大了。why！存疑吧。。。</p>
<p>结合R中的作图，和非常牛逼非常适合统计的<strong>向量数据处理方式</strong>，可以把泊松分布画出来。</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">barplot</span>(<span class="kw">dpois</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">12</span>, <span class="dv">5</span>), <span class="dt">names.arg =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">12</span>, <span class="dt">col =</span> <span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-17"></span>
<img src="M-for-M_files/figure-html/unnamed-chunk-17-1.png" alt="发生1到12次突变的概率分布" width="672" />
<p class="caption">
Figure 2.9: 发生1到12次突变的概率分布
</p>
</div>
</div>
<div id="r" class="section level4">
<h4><span class="header-section-number">2.3.2.2</span> 荡开一会看R语言的函数</h4>
<p>在R语言中，一个概率一般有4种不同的函数，如泊松分布是这样的。</p>
<blockquote>
<ul>
<li>Density, distribution function, quantile function and random generation for the Poisson distribution with parameter lambda.</li>
</ul>
</blockquote>
<table>
<thead>
<tr class="header">
<th>函数</th>
<th>名称</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>dpois(x, lambda, log = FALSE)</strong></td>
<td>密度函数</td>
</tr>
<tr class="even">
<td><strong>ppois(q, lambda, lower.tail = TRUE, log.p = FALSE)</strong></td>
<td>分布函数</td>
</tr>
<tr class="odd">
<td><strong>qpois(p, lambda, lower.tail = TRUE, log.p = FALSE)</strong></td>
<td>分位数函数</td>
</tr>
<tr class="even">
<td><strong>rpois(n, lambda)</strong></td>
<td>随机数生成函数</td>
</tr>
</tbody>
</table>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>参数</th>
<th>意义</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>x</td>
<td>vector of (non-negative integer) quantiles.</td>
</tr>
<tr class="even">
<td>q</td>
<td>vector of quantiles.</td>
</tr>
<tr class="odd">
<td>p</td>
<td>vector of probabilities.</td>
</tr>
<tr class="even">
<td>n</td>
<td>number of random values to return.</td>
</tr>
<tr class="odd">
<td>lambda</td>
<td>vector of (non-negative) means.</td>
</tr>
<tr class="even">
<td>log, log.p</td>
<td>logical; if TRUE, probabilities p are given as log(p).</td>
</tr>
<tr class="odd">
<td>lower.tail</td>
<td>logical; if TRUE (default), probabilities are P[X ≤ x], otherwise, P[X &gt; x].</td>
</tr>
</tbody>
</table>
<blockquote>
<p><strong>quantiles, 分位数</strong>，亦称分位点，是指用分割点将一个随机变量的概率分布范围分为几个具有相同概率的连续区间。分割点的数量比划分出的区间少1，例如3个分割点能分出4个区间。 常用的有中位数、四分位数、十分位数、百分位数等。q-quantile是指将有限值集分为q个接近相同尺寸的子集。</p>
</blockquote>
<p>The Poisson distribution has density</p>
<p><span class="math display">\[p(x) = λ^x e^{-λ}/x!\]</span></p>
<p>for x = 0, 1, 2, … . The mean and variance are E(X) = Var(X) = λ.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">dbinom</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fl">0.5</span>) <span class="co"># 这是在0.5的概率下，扔十次硬币，5次1的概率（假设正面为1，反面为0）。为什么叫density，就是因为只看这一次的，是密度。对应的是massive，质量，在R里面是distribution function, pbinom,就是这个次数之前累积的了。</span></a></code></pre></div>
<pre><code>## [1] 0.2460938</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">pbinom</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="fl">0.5</span>) <span class="co"># </span></a></code></pre></div>
<pre><code>## [1] 0.6230469</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">sum</span>(<span class="kw">dbinom</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">10</span>, <span class="fl">0.5</span>)) <span class="co"># 看这个密度加起来，和上面差不多吧。</span></a></code></pre></div>
<pre><code>## [1] 0.6220703</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">pbinom</span>(<span class="dv">5</span>,<span class="dv">10</span>, <span class="fl">0.5</span> , <span class="dt">lower.tail =</span> <span class="ot">FALSE</span>)  <span class="co"># lower.tail是比这个大</span></a></code></pre></div>
<pre><code>## [1] 0.3769531</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">dpois</span>(<span class="dv">3</span>, <span class="dv">5</span>) <span class="co"># 也是密度</span></a></code></pre></div>
<pre><code>## [1] 0.1403739</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">sum</span>(<span class="kw">dpois</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">5</span>)) <span class="co">#密度之和，</span></a></code></pre></div>
<pre><code>## [1] 0.258288</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">ppois</span>(<span class="dv">3</span>, <span class="dv">5</span>) <span class="co"># distribution</span></a></code></pre></div>
<pre><code>## [1] 0.2650259</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">qpois</span>(<span class="fl">0.2650259</span>, <span class="dv">5</span>) <span class="co"># 按3的概率，倒推出3</span></a></code></pre></div>
<pre><code>## [1] 3</code></pre>
</div>
<div id="section-2.3.2.3" class="section level4">
<h4><span class="header-section-number">2.3.2.3</span> 离散模型的应用</h4>
<p>这里把Bernouli和 Binomial 联系起来说了。Bernouli是单次的是否试验（trail），Binomial是一串Bernouli。抗体可以根据抗原特性折叠，</p>
<p><span class="math display">\[\begin{equation*}
\begin{aligned}
P(X=k) &amp;=&amp; \frac{n\times (n-1)... (n-k+1)}{k\times(k-1)... 1}\; p^k\, (1-p)^{n-k} \\
&amp;=&amp; \frac{n!}{(n-k)!k!}\;p^k\, (1-p)^{n-k}\\
&amp;=&amp; { n \choose k}\; p^k\, (1-p)^{n-k}.\end{aligned}
\end{equation*}\]</span></p>
</div>
</div>
<div id="a-generative-model-for-epitope-detection" class="section level3">
<h3><span class="header-section-number">2.3.3</span> 应用一个表位检测模型（a generative model for epitope detection）</h3>
<p>有的蛋白会引发人体过敏（allergic reaction），蛋白大分子上引发过敏的位置叫表位（epitope）。面对抗原（antigen），白细胞会产生抗体（antibody）。有些抗体直接摧毁抗原，有些召集（recruit）白细胞摧毁抗原。表位是免疫系统识别抗原的关键位置，如抗体、B细胞、T细胞，大多通过表位识别抗原。</p>
<p>基于上述，看这个 <a href="https://en.wikipedia.org/wiki/ELISA">ELISA error model</a>，即<a href="https://zh.wikipedia.org/wiki/%E9%85%B6%E8%81%94%E5%85%8D%E7%96%AB%E5%90%B8%E9%99%84%E8%AF%95%E9%AA%8C">酶联免疫吸附试验</a>，用来检测某蛋白不同位置的特定表位。该模型有以下假定：</p>
<blockquote>
<ul>
<li></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li>在100个不同位置检测某蛋白，检测相对独立。</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li>本章检测50个病人的样本。</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;data/e100.RData&quot;</span>)</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">e100</a></code></pre></div>
<pre><code>##   [1] 2 0 1 0 0 0 2 1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 1 0 1 0 0 0 1 0 0 0 0 1 0
##  [36] 1 1 0 1 2 2 7 1 0 2 0 1 0 1 1 1 1 0 1 0 0 0 0 1 2 2 1 0 0 0 0 0 1 0 0
##  [71] 0 1 1 0 0 1 1 0 0 0 0 1 0 0 0 1 0 1 0 0 1 1 0 0 1 1 0 0 1 0</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw">barplot</span>(e100, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">7</span>), <span class="dt">width =</span> <span class="fl">0.7</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">100.5</span>),</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">  <span class="dt">names.arg =</span> <span class="kw">seq</span>(<span class="dt">along =</span> e100), <span class="dt">col =</span> <span class="st">&quot;darkolivegreen&quot;</span>)</a></code></pre></div>
<p><img src="M-for-M_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="statistical-modeling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ecowang/MforM/edit/master/01-generative.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["M-for-M.pdf", "M-for-M.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
